src/crontab.c

```diff
@@ -409,8 +413,17 @@ static void list_cmd(void) {
 	/* file is open. copy to stdout, close.
 	 */
 	Set_LineNum(1)
-		while (EOF != (ch = get_char(f)))
+	while (EOF != (ch = get_char(f))){
+		if(new_line){
+			if(ch == '#' && isatty(STDOUT)){
```

Comment:
Call the isatty() before the cycle. And all the coloring directive output must be properly guarded by this.

Reply:
@t8m I'm unsure what you mean. Do you suggest I modify these lines to:
```C
		if (isatty(STDOUT) && new_line) {
			if (ch == '#') {
```

I agree this version is better than the previous one.

Reply:
Yes, move the condition there but also call the function before the cycle and save the isatty() result to a variable.

Reply:
Thank you for the explanation!
I committed the changes into commit https://github.com/cronie-crond/cronie/pull/39/commits/a490da1a86c3c89c1379f138fa04bccc787c7462